guides/common/modules/proc_configuring-logging-to-journal-or-file-based-logging.adoc

```diff
@@ -0,0 +1,42 @@
+[id="Configuring_Logging_to_Journal_or_File-based_Logging_{context}"]
+= Configuring Logging to Journal or File-based Logging
+
+{Project} uses file-based logging by default.
+You can use the `{foreman-installer}` command to reconfigure logging.
+
+.Procedure for configuring logging with Journal
+* Enter the following `{foreman-installer}` command to configure logging to the `journald` service:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# {foreman-installer} \
+--foreman-logging-layout pattern \
+--foreman-logging-level info \
+--foreman-logging-type journald \
+--foreman-proxy-log JOURNAL
+----
+
+Journal now forwards log messages to `rsyslog`, and `rsyslog` writes the log messages to `/var/log/messages`.
+After this change, the log messages do not appear in `/var/log/foreman/production.log` or `/var/log/foreman-proxy.log`.
+
+.Procedure for configuring file-based logging
+* Enter the following `{foreman-installer}` command to configure the default file-based logging:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# {foreman-installer} \
+--reset-foreman-logging-layout \
+--reset-foreman-logging-level \
+--reset-foreman-logging-type \
+--reset-foreman-proxy-log
+----
+
+After this change, the log messages appear in `/var/log/foreman/production.log` or `/var/log/foreman-proxy.log`.
+
+ifdef::satellite[]
+.Additional resources
+For more information about Journal, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/configuring_basic_system_settings/index#viewing-logs-using-the-command-line_assembly_troubleshooting-problems-using-log-files[Viewing logs using the command line] in the _{RHEL} 8 Configuring Basic System Settings Guide_.
+endif::[]
+ifndef::satellite,orcharhino[]
```

Comment:
Even `orcharhino` should have more information on Journal, No?

Reply:
I think this is a question for @maximiliankolb, is that right? Do you want to add additional resources on Journal while I'm working on this module?

Reply:
I think it's not the worst thing to do: avoid linking to some random GH repo from some (former) developer. The information there is getting outdated. It may be worth incorporating https://github.com/lzap/foreman-elasticsearch#system-journal-metadata into official docs.

Reply:
@asteflova orcharhino generally does not link to access.redhat.com.

Looking at the commit messages in lzap's repo, I'd also not link to this for any build target. Maybe Ewoud knows if/what to take from there and incorporate here in our own words?

Reply:
I removed the link to https://github.com/lzap/foreman-elasticsearch . Right now, the only additional resource listed is access.redhat.com for satellite builds.

If anyone wants to add another link or include more information into the procedure itself, let me know. (Note that I will be adding more details on inspecting the logs based on another of Ewoud's comment.) Let's just agree not to overdo it :) No need to add something just for the sake of adding it.

Reply:
I think the table of metadata that you can filter on is useful.

The list of loggers is actually more complete than what we have in https://github.com/theforeman/foreman-documentation/blob/master/guides/common/modules/proc_enabling-individual-loggers.adoc. Note that https://github.com/theforeman/foreman-documentation/blob/a56dfe21b2c9f640d54da81cced6f1e4ca43e06c/guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc#L109 roughly describes the same thing, but in a way that we shouldn't support.

Reply:
Thanks everyone for chiming in. This is what I intend to do:

* Add [System journal metadata](https://github.com/lzap/foreman-elasticsearch#system-journal-metadata) to the guide (probably as a reference module)
* Cross-check the list of logger names in [System journal metadata](https://github.com/lzap/foreman-elasticsearch#system-journal-metadata) with [Enabling individual loggers](https://github.com/theforeman/foreman-documentation/blob/master/guides/common/modules/proc_enabling-individual-loggers.adoc), add whatever is missing to Enabling individual loggers

@ekohl When you write that we shouldn't support [Increasing the Logging Level On {Project}](https://github.com/theforeman/foreman-documentation/blob/a56dfe21b2c9f640d54da81cced6f1e4ca43e06c/guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc?plain=1#L109), does that mean I should just remove the whole section? (I'm fine with it, I'm just checking I understand correctly.)

Reply:
> @ekohl When you write that we shouldn't support [Increasing the Logging Level On {Project}](https://github.com/theforeman/foreman-documentation/blob/a56dfe21b2c9f640d54da81cced6f1e4ca43e06c/guides/common/modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc?plain=1#L109), does that mean I should just remove the whole section? (I'm fine with it, I'm just checking I understand correctly.)

I meant that we shouldn't support it this way. Manually editing files is something we shouldn't do, but rather rely on the installer to do so. I started https://github.com/theforeman/foreman-documentation/pull/1715 but that should be revisited. I'd be happy if if anyone has time to take that over and complete it.

Reply:
I can take a look at #1715 later this week to see if it's something I could do while I'm looking into logging anyway.

For now, I'll resolve this conversation: I added the system journal metadata table and the individual loggers from lzap's repo.